assets/source/setup-guide/app/helpers/documentation-link-props.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * External dependencies
+ */
+import { recordEvent } from '@woocommerce/tracks';
+
+/**
+ * Clicking on an external documentation link.
+ *
+ * @event wcadmin_pfw_documentation_link_click
+ *
+ * @property {string} link_id Identifier of the link.
+ * @property {string} context What action was initiated.
+ * @property {string} href Href to which the user was navigated to.
+ */
+
+/**
+ * Creates properties for an external documentation link.
+ * May take any other props to be extended and forwarded to a link element (`<a>`, `<Button isLink>`).
+ *
+ * Sets `target="_blank" rel="noreferrer"` and onClick handler that fires track event.
+ *
+ * @fires pfw_documentation_link_click on click, with given `linkId` and `context`.
+ *
+ * @param {Object} props React props.
+ * @param {string} props.href Href to used by link and in track event.
+ * @param {string} props.linkId Forwarded to {@link wcadmin_pfw_documentation_link_click}
+ * @param {string} props.context Forwarded to {@link wcadmin_pfw_documentation_link_click}
+ * @param {string} [props.target='_blank']
+ * @param {string} [props.rel='noreferrer']
+ * @param {Function} props.onClick
+ * @param {...import('react').AnchorHTMLAttributes} props.props
+ */
+export default function documentationLinkProps( {
+	href,
+	linkId,
+	context,
+	target = '_blank',
+	rel = 'noreferrer',
+	onClick,
+	...props
+} ) {
+	return {
+		href,
+		target,
+		rel,
+		...props,
+		onClick: ( event ) => {
+			if ( onClick ) {
+				onClick( event );
+			}
```

Comment:
📜 
A little bit of concern here when using this helper like this:

```js
// Case A
const buttonA = (
	<Button
		isLink
		onClick={ () => { /* do something */ } }
		{ ...documentationLinkProps( {
			// ...omitted
		} ) }
	>
		Hi there
	</Button>
);

// Case B
const buttonB = (
	<Button
		isLink
		{ ...documentationLinkProps( {
			// ...omitted
		} ) }
		onClick={ () => { /* do something */ } }
	>
		Hi there
	</Button>
);
```

The `onClick` handler of **Case A** will be overwritten and not be called. And the handler of **Case B** will overwrite the tracking event handler. And these potential issues cannot be checked by codes or lint tools.
- For Case A, it's easy to find during development, but it could happen accidentally when moving code.
- For Case B, it's easy to miss that tracking event is overwritten and not triggered, after all, tracking occurs in the background and probably is not set `localStorage.setItem('debug', 'wc-admin*')` or DevTool Console does not monitor the verbose log.

However, considering that links using this helper usually don't have an additional `onClick` handler, perhaps it would be better to enhance the use of JSDoc regarding `onClick`.

Reply:
Thanks for catching that! IMHO, silent future bugs are a quite serious thing.
Do you have any idea how to solve that?

I'd try to emphasize that somewhat in the docs, and change the usage in this PR to 
```jsx
<Button
{ ...documentationLinkProps( {
	href:
		wcSettings.pinterest_for_woocommerce
			.pinterestLinks.adGuidelines,
	linkId: 'ad-guidelines',
	context: view,
	rel: 'noopener',
	// Other Button props.
	isLink,
} ) }>
```
To somewhat suggest that it's safer to pass all the props through the helper.

Reply:
> Do you have any idea how to solve that?

I haven't used the "Component hooks" pattern in practice, my opinion is just based on my imagination and empirical inference, hope it could still bring some idea though.

The pattern looks more likely to be used in a fundamental components library. These core components might share a lot of reusable or composable props. Thus, with this pattern, it creates default props for components that are designed to be overridden if there are other props specified after its sequence.

For example, a library provides a `<RichTextarea>` that has a default `onPaste` handler for accepting Markdown syntax, and the default handler is composited by a bunch of Component hooks. And the docs would indicate that if library users set `onPaste` on it, that means they want to handle the paste event themself instead of the default handler. And if library users need to append their handler to pre/post-process something, I would design other dedicated props for that and pass them to internal `useMyAwesomeMarkdownParser()` to merge events.

But in our application-level plugins development, this pattern does not seem to work so smoothly. Since when using React, if there is no particular reason, the props object spreading is usually placed at the first because the convention is that they are fine to be overridden. And props that need to be synthesized would be processed by separate logic.

Back to this case. It's a bit tricky because we don't expect the tracking calling in `onClick` to be covered, but in the meantime, these tracking usages would probably not be added with additional `onClick` callbacks either. Probably all of them are working well with the current implementation in the future, eventually. But the problem is as you said "silent future bugs are a quite serious thing", which I totally agree with it. I've been thinking about this issue today till now, but I don't think of a solution based on current this pattern. So I'd suggest considering another way if we still have no idea then.